File: 2090-KRadiusSubarrayAverages.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        ws = 2*k + 1
        n = len(nums)
        s = 0
        result = [-1] * n
        
        for i in range(n):
            s += nums[i]
            if i >= ws - 1:
                result[i - (ws//2)] = s // ws
                s -= nums[i - ws + 1]

        return result

File: 2090-KRadiusSubarrayAverages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        int n = nums.size();
        int windowSize = 2 * k + 1;
        vector<int> result(n, -1);
        long long total = 0;

        for(int i = 0; i < n; i++){
            total += nums[i];
         
            if(i >= windowSize - 1){
                result[i - windowSize / 2] = total / windowSize;
                total -= nums[i - windowSize + 1];
            }
        }

        return result;
    }
};